manifests/plugin.pp

```diff
@@ -32,8 +32,22 @@
 ) {
   include ::jenkins::params
 
-  $plugin            = "${name}.hpi"
-  $plugin_parent_dir = inline_template('<%= @plugin_dir.split(\'/\')[0..-2].join(\'/\') %>')
+  $plugin     = "${name}.hpi"
+  $homedir    = $::jenkins::params::homedir
+  $jenkinsdir = $::jenkins::params::jenkinsdir
+  $jusername = $username ? {
+    ''      => $::jenkins::params::jenkins_user,
+    default => $username,
+  }
+  $jgroup = $group ? {
+    ''      => $::jenkins::params::jenkins_group,
+    default => $group,
+  }
+  $jplugin_dir = $plugin_dir ? {
+    ''      => $::jenkins::params::plugin_dir,
+    default => $plugin_dir,
+  }
```

Comment:
I figured that I can do something like:

define jenkins::plugin (
foo,
) {
Class['jenkins::params'] ->
Jenkins::Plugin[$name]
}

to ensure that the jenkins::params are evaluated before the define.

See PR #356 , first start of to configure the user/group in params.pp